transport/TransportFactory.h

```diff
@@ -67,6 +68,12 @@ class TransportFactory
     virtual bool openRtpMuxPorts(Endpoints& rtpPorts) const = 0;
 
     virtual void maintenance(uint64_t timestamp) = 0;
+
+    virtual void registerProbeServer(ProbeServer& server) = 0;
```

Comment:
You can fix it later, but I think TransportFactory can be unaware of the ProbeServer type. 
Just take the Endpoint::IEvents and ServerEndpoint::IEvents interfaces. Perhaps even two register methods, one for each interface.

Reply:
We need to provide ICE credentials when registering a listener for endpoints, as `registerListener(const std::string& stunUserName, IEvents* listener)`. I opted to have them in ProbeServer, and let it accept endpoints for registration, hence TransportFactory knows about the ProbeServer.

I can change it and move static ICE credentials to TransportFactory also.